### DISCUSSION TITLE
getInitialProps does not pass props to component

### DISCUSSION THREAD

**Discussion by nmtri-dev** [vercel/next.js#17224] [Help] [answered]
I have 1 page like this
```
const Index = (props) => {
  return <Layout page='home'>
    {console.log(222, props)}
    <Head>
      {...}
    </Head>
    <Homepage clips={props.clips} />
  </Layout>;
}

Index.getInitialProps = async () => {
  const res = await fetch(
    `...`
  )

  const json = await res.json();
  console.log(111, json);
  return {
    clips: json
  }
}

export default Index
```
And this is what I got from server log
```
111 [ { _id: '5f65f382a98ecd003d5ca53e',
    ...' } ]
222 {}
```
---
NextJS version `9.1.5`
This was working well for me til I restructure project and have docker for the deployment.

Anyone has an idea?

**Comment by jamesmosier** [comment, marked as answer]
Hi @tri-nguyen-nfq. Do you have a custom `_app.js`? If so, could you share the code?

**Comment by nmtri-dev** [comment]
Thank you @jamesmosier , you guess it right, I mess up the custom `_app.js`

**Comment by joshverd** [comment]
In case someone comes across this in the future and is wondering what the solution is, make sure you're passing the pageProps into the returned component in `_app.tsx`'s render method.

```ts
// Type Imports
import type { AppProps } from 'next/app';

const AppComponent = (props: AppProps) => {
  return (
    <props.Component {...props.pageProps} />
  );
};

export default AppComponent;
```
